src/app/modules/pages/cart/cart.component.ts

```diff
@@ -0,0 +1,51 @@
+import {
+  ChangeDetectionStrategy,
+  Component,
+  OnDestroy,
+  OnInit,
+} from '@angular/core';
+import { IProduct } from '@interfaces/product.interface';
+import { CartService } from '@pages/cart/cart.service';
+import { Subscription } from 'rxjs';
+
+@Component({
+  selector: 'app-cart',
+  templateUrl: './cart.component.html',
+  styleUrls: ['./cart.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class CartComponent implements OnInit, OnDestroy {
+  public cartProducts: IProduct[] = [];
+
+  private subscription!: Subscription;
+
+  constructor(private cartService: CartService) {}
+
+  public changeCount(product: IProduct, value: number) {
+    console.log(product.id);
+    console.log(value);
+
+    this.cartService.setCount(product.id, value);
+  }
+
+  public getCount(product: IProduct): number {
+    return this.cartService.getCount(product.id);
+  }
+
+  public getTotalPrice(cartProducts: IProduct[]) {
+    return this.cartService.getTotalPrice(cartProducts);
+  }
+
+  ngOnInit(): void {
+    this.cartService.getFromStorage();
+    this.subscription = this.cartService.cartChanged$.subscribe(
```

Comment:
You could also try using an async pipe, but this solution is fine

Reply:
In this case i need to use public access modifier to get stream in html
Private services cant be used in template
Is it a good practice to change access modifier?

Reply:
Instead of a separate variable in the service, you can make a method that will return "Observable", and put it into a public variable, and in the future process it in html via "async" pipe, it is good practice, because you do not need to write extra code like "subscriptions", and add extra lifecycle method (onDestroy), "async" itself will unsubscribe from the thread.